### PR TITLE
feat: autonomy engine phase 2 — hard gates and action_risk enforcement

### LINKED ISSUE #147: feat: autonomy engine phase 2 — hard gates and action_risk enforcement
## Background

Phase 1 established the global autonomy score, CEO controls (`get-autonomy` / `set-autonomy`), and prompt injection. The `action_risk` field was added to `skill.json` manifests as documentation-only — the execution layer does not yet enforce it.

Phase 2 wires the gates. The intended design is documented in `docs/superpowers/specs/2026-04-03-autonomy-engine-design.md` under "Phase 2: Hard Gates".

---

### 1. Make `action_risk` a required field

- Skill load (`SkillRegistry.register`) must reject any manifest missing `action_risk` with a startup error
- `SkillManifest` TypeScript type: move `action_risk` from optional to required
- Any existing skills missing the field must be patched as part of this PR

### 2. Execution-layer gate

When an agent invokes a skill, the execution layer compares `AutonomyService.minScoreForActionRisk(manifest.action_risk)` against the live score from `autonomy_config`. If the score is below the threshold:

- Return `{ success: false, error: "..." }` to the agent with an advisory message explaining why the skill was blocked and what autonomy score is required
- Emit an audit event: `autonomy.skill_blocked` with `{ skill, action_risk, current_score, required_score }`
- Do **not** throw — existing skill error handling applies

### 3. Outbound gateway gate (`score < 70`)

`OutboundGateway` is already the choke point for all outgoing email. Add a check:

- If `autonomy_config.score < 70`: block dispatch and return an advisory to the agent; do not send
- This gate applies even if the skill itself passes the action_risk check (belt-and-suspenders for `medium` skills)
- Emit `autonomy.email_blocked` on the bus

### 4. Full-restriction gate (`score < 60`)

- All skill invocations in an `agent.task` context return early with an advisory if `score < 60`, regardless of the skill's `action_risk`
- Read-only skills (`action_risk: "none"`) are exempt — they always proceed

### 5. Approval-event mechanism (stretch, can be a follow-up)

For the `score < 70` / outbound case, provide a path for the CEO to approve a specific pending action inline without changing the global score. Design TBD.

---

## Acceptance Criteria

- [ ] `SkillRegistry.register` rejects manifests with missing `action_risk` at startup
- [ ] All existing skills in `skills/` have `action_risk` declared
- [ ] Execution layer enforces action_risk threshold against live autonomy score
- [ ] Outbound gateway enforces score < 70 gate independently
- [ ] Score < 60 blocks all non-read skills with advisory return
- [ ] Blocked invocations emit audit events
- [ ] Unit tests for all three gate paths (pass, fail-advisory, full-restriction)
- [ ] Phase 2 design doc filed at `docs/superpowers/specs/`

---

## Notes

`AutonomyService.minScoreForActionRisk()` already exists and is correct — the execution layer just needs to call it. The `autonomy_config` read is already one-per-task in the coordinator; the execution layer will need its own read or the score passed down via context.

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

Closes #147

Wires hard enforcement gates into the execution layer and outbound gateway so the autonomy score is no longer advisory-only. Three gate layers enforce `action_risk` thresholds at runtime:

- **Gate A (score < 60):** Blocks all non-read skills (`action_risk !== 'none'`) when the agent is in restricted mode
- **Gate B (per-skill threshold):** Compares the live autonomy score against `AutonomyService.minScoreForActionRisk(manifest.action_risk)` and blocks if below threshold
- **Gateway gate (score < 70):** Blocks outbound sends (`OutboundGateway.send()`) — drafts remain unaffected as the intended fallback at lower autonomy levels

Key design decisions:
- **Fail-open:** If `AutonomyService` is absent or `getConfig()` returns null/throws, gates are skipped with a warn log (behavioral guardrail, not security boundary)
- **Per-invocation score reads:** Each `invoke()`/`send()` reads the score independently — no caching, always uses the live value
- **Elevated skill exemption:** Skills with `sensitivity: 'elevated'` bypass autonomy gates entirely (prevents `set-autonomy` deadlock)
- **`action_risk` required at startup:** `SkillRegistry.register()` now rejects manifests missing the field (runtime matches the TypeScript type)
- **Bus events:** `autonomy.skill_blocked` and `autonomy.send_blocked` emitted as fire-and-forget audit events

21 files changed, +1870/-30 lines. 1698 tests passing, typecheck clean.

## Test plan

- [ ] `npm test -- src/skills/execution.test.ts` — 10 new autonomy gate tests (threshold blocking/passing, none exemption, full restriction, event emission, fail-open variants, elevated skill exemption)
- [ ] `npm test -- tests/unit/skills/outbound-gateway.test.ts` — 6 new tests (score < 70 blocks send, score >= 70 allows, event emission, fail-open, draft not gated, DB error path)
- [ ] `npm test -- tests/unit/skills/registry.test.ts` — existing + new test for missing action_risk rejection
- [ ] `npm test` — full suite (1698 tests passing, no regressions)
- [ ] `npx tsc --noEmit` — typecheck clean
- [ ] Verify existing integration tests still pass with action_risk added to test manifests

## Release Notes

* **New Features**
  * Skill invocations are now blocked when the live autonomy score falls below required thresholds.
  * Direct outbound sends are blocked when autonomy score is below 70.
  * Audit events are emitted when autonomy-based blocks occur.

* **Chores**
  * Autonomy requirements for skills are now strictly enforced at startup.
  * Updated documentation to reflect Phase 2 implementation status.

### CODE REVIEW

**Comment by josephfung** [comment]
@coderabbitai code review please

File: docs/specs/14-autonomy-engine.md [hunk 1]
```diff
@@ -9,7 +9,7 @@
 A single global autonomy score (0–100) governs how independently Curia operates across all agents and skills. The score is CEO-controlled, Postgres-persisted, and injected into the Coordinator's system prompt on every task — so Curia self-governs in real time without a restart.
 
 **Phase 1 (implemented):** Score storage, CEO read/write skills, behavioral prompt injection.
-**Phase 2 (future):** Hard execution gates — skill invocations blocked when score is below the skill's declared `action_risk` floor.
+**Phase 2 (implemented):** Hard execution gates — skill invocations blocked when score is below the skill's declared `action_risk` floor.
```
[the remark this answers is not in the document]
Comment (josephfung) [reply docs/specs/14-autonomy-engine.md:12]: Good catch — the top-level status said "Implemented (Phase 1)" while the body already said Phase 2 was implemented. Updated to "Implemented (Phase 1 & 2)" in 7478d0c.

**Comment by josephfung** [comment]
All four @coderabbitai findings addressed in 7478d0c:

1. **Spec status header** (inline) — Updated from "Implemented (Phase 1)" to "Implemented (Phase 1 & 2)". ✅

2. **CHANGELOG duplicate headings** (outside diff) — Consolidated into single `### Added`, `### Changed`, `### Fixed` sections under `## [Unreleased]`. ✅

3. **Phase 2 section future tense / approval event** (outside diff) — Changed "When Phase 2 is implemented, these are the first gate candidates:" → "Phase 2 enforces three hard gates:". Replaced "requires an explicit approval event" with actual behavior: "blocks direct outbound dispatch; draft creation remains available as the intended fallback". Also fixed the implementation status table row. ✅

4. **Missing null-config test** (nitpick) — Added `fails open when getConfig returns null (pre-migration)` test to the outbound gateway suite. Asserts `result.success === true` and no `autonomy.send_blocked` event emitted. ✅

File: docs/specs/14-autonomy-engine.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (josephfung) [reply docs/specs/14-autonomy-engine.md:12]: @coderabbitai any questions or is this okay?
